### PR TITLE
handle different arity of type params for interfaces

### PR DESCRIPTION
**Pull request by awareness481**
**Description:**

Ref: https://github.com/dudykr/stc/issues/956

### CODE REVIEW

File: crates/stc_ts_file_analyzer/src/analyzer/decl_merging.rs [hunk 1]
```diff
@@ -97,24 +97,37 @@ impl Analyzer<'_, '_> {
...
-                // TODO: Handle the number of type parameters.
                 let mut type_params = FxHashMap::default();
-                if let Some(b_tps) = &bi.type_params {
-                    if let Some(a_tp) = &a.type_params {
+
+                match (&a.type_params, &bi.type_params) {
+                    (Some(a_tps), Some(b_tps)) => {
+                        if a_tps.params.len() != b_tps.params.len() {
+                            self.storage
+                                .report(ErrorKind::InterfaceNonIdenticalTypeParams { span: a.span() }.into());
```
Comment (awareness481) [inline crates/stc_ts_file_analyzer/src/analyzer/decl_merging.rs:106]: span currently looks like this
```rs
interface A<T extends string> {}
────────────────────────────────
```
is there any way to only use the span of the name of the interface? (like tsc)

```rs
```rs
interface A<T extends string> {}
          ─
```
```

File: crates/stc_ts_file_analyzer/src/analyzer/decl_merging.rs [hunk 2]
```diff
@@ -97,24 +97,37 @@ impl Analyzer<'_, '_> {
...
+                        }
+
                         for (idx, b_tp) in b_tps.params.iter().enumerate() {
                             type_params.insert(
                                 b_tp.name.clone(),
                                 Type::Param(TypeParam {
-                                    span: a_tp.span,
-                                    name: a_tp.params[idx].name.clone(),
+                                    span: a_tps.span,
+                                    name: a_tps.params.get(idx).unwrap_or(b_tp).name.clone(),
```
Comment (awareness481) [inline crates/stc_ts_file_analyzer/src/analyzer/decl_merging.rs:116]: Does it matter here that `a_tps.span` can be passed when `a_tps.params.get(idx) = None` ?

File: crates/stc_ts_file_analyzer/src/analyzer/decl_merging.rs (hunk 1 above)
Comment (kdy1) [reply crates/stc_ts_file_analyzer/src/analyzer/decl_merging.rs:106]: I think we should add a field like `name_span` to the `Interface` type

File: crates/stc_ts_file_analyzer/src/analyzer/decl_merging.rs (hunk 2 above)
Comment (kdy1) [reply crates/stc_ts_file_analyzer/src/analyzer/decl_merging.rs:116]: Hmm, not sure, but I don't think it matters.

**Review by kdy1 — APPROVED** [review]
Thank you!

**Pull request merged**